text-embeddings-inference/README.md

```diff
@@ -0,0 +1,14 @@
+## Docker build
+
+```shell
+docker build . -f Dockerfile-hpu -t text-embeddings-inference:hpu-0.6
+```
+
+## Docker run
+
+```shell
+model=sentence-transformers/all-MiniLM-L6-v2
+volume=$PWD/data # share a volume with the Docker container to avoid downloading weights every run
+
+docker run --runtime=habana -e HABANA_VISIBLE_DEVICES=all -e OMPI_MCA_btl_vader_single_copy_mechanism=none --cap-add=sys_nice --net=host --ipc=host -p 8080:80 -v $volume:/data text-embeddings-inference:hpu-0.6 --model-id $model --pooling cls
```

Comment:
seems to work for me when I remove --net=host. Should we remove it?

otherwise I see "WARNING: Published ports are discarded when using host network mode" and cannot send requests such as

`curl 127.0.0.1:8080/embed \
    -X POST \
    -d '{"inputs":"What is Deep Learning?"}' \
    -H 'Content-Type: application/json'`

Reply:
I have taken it from official Habana installation guide - https://docs.habana.ai/en/latest/Installation_Guide/Bare_Metal_Fresh_OS.html#setup-ami-for-container-bare:

>      docker run -it --runtime=habana -e HABANA_VISIBLE_DEVICES=all -e OMPI_MCA_btl_vader_single_copy_mechanism=none --cap-add=sys_nice --net=host --ipc=host vault.habana.ai/gaudi-docker/1.14.0/{$OS}/habanalabs/pytorch-installer-2.1.1:latest

I think it is better to keep TEI docker consistent with the installation guide. If the guide will remove `--net=host` then we can remove it from TEI docker.

What do you think?

Reply:
Regarding calling /embed endpoint. There are two options:
* With `--net=host` we should call `127.0.0.1:80/embed`
* Without `--net=host` we should call `127.0.0.1:8080/embed`

Reply:
To keep the HPU support consistent with the rest of TEI images I have removed `--net=host`, so user can run exactly the same command as in TEI documentation:

    curl 127.0.0.1:8080/embed \
      -X POST \
      -d '{"inputs":"What is Deep Learning?"}' \
      -H 'Content-Type: application/json'

@skaulintel thanks for bringing it up!